Bonsai.Configuration.Tests/ScriptExtensionsProjectMetadataTests.cs

```diff
@@ -0,0 +1,380 @@
+          <PropertyGroup>
+            <TargetFramework>net472</TargetFramework>
+          </PropertyGroup> 
+          <ItemGroup>
+            <PackageReference Include="FakePackageA" Version="3226.42.1337" />
+            <PackageReference Include="FakePackageB" Version="126.484.30" />
+          </ItemGroup>
+        </Project>
+        """, LoadOptions.PreserveWhitespace);
+        var metadata = new ScriptExtensionsProjectMetadata(document);
+        metadata = metadata.AddPackageReferences([new PackageReference("FakePackageA", "3226.42.1338")]);
+        var xml = metadata.GetProjectXml();
+        var xElement = XElement.Parse(xml);
+        var itemGroups = xElement.Descendants("ItemGroup").ToArray();
+        var packageReferences = xElement.Descendants("PackageReference").ToArray();
+
+        CollectionAssert.AreEquivalent((string[])["FakePackageA", "FakePackageB"], metadata.GetPackageReferences().ToArray());
+
+        // Package was udpated
+        var packageAs = xElement.XPathSelectElements("/ItemGroup/PackageReference[@Include='FakePackageA']").ToArray();
+        Assert.AreEqual(1, packageAs.Length);
+        Assert.AreEqual("3226.42.1338", packageAs[0].Attribute("Version").Value);
+        Assert.AreEqual(-1, xml.IndexOf("3226.42.1337"));
+
+        // Unrelated package untouched
+        var packageBs = xElement.XPathSelectElements("/ItemGroup/PackageReference[@Include='FakePackageB']").ToArray();
+        Assert.AreEqual(1, packageBs.Length);
+        Assert.AreEqual("126.484.30", packageBs[0].Attribute("Version").Value);
+
+        // Only the two packages are present
+        Assert.AreEqual(2, xElement.Descendants("PackageReference").Count());
+    }
+
+    [TestMethod]
+    public void DowngradeExistingPackageReference()
+    {
+        var document = XDocument.Parse
+        ($"""
+        <Project Sdk="Microsoft.NET.Sdk">            
+          <PropertyGroup>
+            <TargetFramework>net472</TargetFramework>
+          </PropertyGroup> 
+          <ItemGroup>
+            <PackageReference Include="FakePackageA" Version="3226.42.1337" />
+            <PackageReference Include="FakePackageB" Version="126.484.30" />
+          </ItemGroup>
+        </Project>
+        """, LoadOptions.PreserveWhitespace);
+        var metadata = new ScriptExtensionsProjectMetadata(document);
+        metadata = metadata.AddPackageReferences([new PackageReference("FakePackageA", "3226.42.1336")]);
+        var xml = metadata.GetProjectXml();
+        var xElement = XElement.Parse(xml);
+        var itemGroups = xElement.Descendants("ItemGroup").ToArray();
+        var packageReferences = xElement.Descendants("PackageReference").ToArray();
+
+        CollectionAssert.AreEquivalent((string[])["FakePackageA", "FakePackageB"], metadata.GetPackageReferences().ToArray());
+
+        // Package was not updated
+        //TODO: This is how the implementation is written, but is this behavior correct?
```

Comment:
This test covers the behavior provided by this check:

https://github.com/bonsai-rx/bonsai/blob/5279025d323e4e7f01a7de373fca4627206a0430/Bonsai.Configuration/ScriptExtensionsProjectMetadata.cs#L98

It's been like this since 2018 [when automatic `Extensions.csproj` updating was introduced](https://github.com/bonsai-rx/bonsai/commit/e1ba312c6950bff92101f39469b86f2ee427d308#diff-506e74263843b950cb380fa7720d14254eb9fe7cd60b3941dd8a06494a8aabc5R73).

Is this behavior actually a good idea? It seems to me like the `Extensions.csproj` should always match the Bonsai environment since the two not matching only causes confusion.

Reply:
Created an issue to follow up on this: https://github.com/bonsai-rx/bonsai/issues/2275